llvm/lib/MC/MachObjectWriter.cpp

```diff
@@ -1097,10 +1094,11 @@ void MachObjectWriter::writeDataInCodeRegion(MCAssembler &Asm) {
     else
       report_fatal_error("Data region not terminated");
 
-    LLVM_DEBUG(dbgs() << "data in code region-- kind: " << Data.Kind
-                      << "  start: " << Start << "(" << Data.Start->getName()
-                      << ")" << "  end: " << End << "(" << Data.End->getName()
-                      << ")" << "  size: " << End - Start << "\n");
+    LLVM_DEBUG(dbgs() << "data in code region-- kind: "
+                      << (MachO::DataRegionType)Data.Kind << "  start: "
```

Comment:
Why this diff?

Reply:
To make layering violations less apparent (even though I will rewrite MachOCASWriter to be a subclass of MachObjectWriter), I changed
```
  struct DataRegionData {
    MachO::DataRegionType Kind;
    MCSymbol *Start;
    MCSymbol *End;
  }; 
```
to
```
  struct DataRegionData {
    unsigned Kind;
    MCSymbol *Start;
    MCSymbol *End;
  };
```
so to properly print out the `Kind` in the `LLVM_DEBUG` block, I had to cast to `MachO::DataRegionType`